### PR TITLE
Fix information endpoint add test for endpoints

### PR DESCRIPTION
**Pull request by a-breid-ericsson**
### Applicable Issues
/information endpoint was not working and caused an error.

### Description of the Change
Fix bug that made /information endpoint stop working.
Add functional tests for all endpoints

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: src/functionaltests/resources/features/restEndpoints.feature [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    Then Request should get response code <responsecode>
+
+  Examples: 
+    | responsecode | type   | add_sub | endpoint                         |
+    | 200          | POST   | do      | /subscriptions                   |
+    | 200          | PUT    | do      | /subscriptions                   |
+    | 200          | GET    | do not  | /subscriptions/test_subscription |
+    | 200          | DELETE | do not  | /subscriptions/test_subscription |
+    | 404          | GET    | do not  | /subscriptions/test_subscription |
+
```
Comment (a-breid-ericsson) [inline src/functionaltests/resources/features/restEndpoints.feature]: Note that the Scenario "Test POST PUT GET DELETE on /subscriptions endpoint" does the same test as the Scenario Outline "Perform <type> request on <endpoint> and expect response code <responsecode>" discussion what to keep should be done.

Comment (Christoffer-Cortes) [reply src/functionaltests/resources/features/restEndpoints.feature]: I think both are fine. Pick one that suits you.

File: src/functionaltests/java/com/ericsson/ei/restendpoints/RestEndpointsTestSteps.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+    @When("^Perform request on endpoint \"([^\"]*)\"$")
+    public void perform_request_on_endpoint(String endpoint) throws Throwable {
+        System.out.println("############# Endpoint ::: <" + endpoint + ">");
+        response = request.setEndpoint(endpoint)
+                          .performRequest();
+    }
+
+    @Then("^Request should get response code (\\d+)$")
+    public void request_should_get_response_code(int expectedstatusCode) throws Throwable {
```
Comment (e-edling-ericsson) [inline src/functionaltests/java/com/ericsson/ei/restendpoints/RestEndpointsTestSteps.java]: capital s on status :)

File: src/functionaltests/java/com/ericsson/ei/restendpoints/RestEndpointsTestSteps.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    }
+
+    @Given("^A DELETE request is prepared$")
+    public void a_DELETE_request_is_prepared() {
+        request = new HttpRequest(HttpMethod.DELETE);
+        setRequestDefaults();
+    }
+
+    @Given("^Subscription with name \"([^\"]*)\" is added as body$")
+    public void subscription_with_name_is_added_as_body(String name) throws Throwable {
```
Comment (e-edling-ericsson) [inline src/functionaltests/java/com/ericsson/ei/restendpoints/RestEndpointsTestSteps.java]: i would change the "name" to "subscriptionName" to reflect more what it is

File: src/functionaltests/java/com/ericsson/ei/restendpoints/RestEndpointsTestSteps.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    @When("^Perform request on endpoint \"([^\"]*)\"$")
+    public void perform_request_on_endpoint(String endpoint) throws Throwable {
+        System.out.println("############# Endpoint ::: <" + endpoint + ">");
+        response = request.setEndpoint(endpoint)
+                          .performRequest();
+    }
+
+    @Then("^Request should get response code (\\d+)$")
+    public void request_should_get_response_code(int expectedstatusCode) throws Throwable {
+        System.out.println("############# expectedstatusCode ::: " + expectedstatusCode);
```
Comment (e-edling-ericsson) [inline src/functionaltests/java/com/ericsson/ei/restendpoints/RestEndpointsTestSteps.java]: Remove sysout or use logger i suppose?

File: src/functionaltests/java/com/ericsson/ei/restendpoints/RestEndpointsTestSteps.java (hunk 2 above)
Comment (a-breid-ericsson) [reply src/functionaltests/java/com/ericsson/ei/restendpoints/RestEndpointsTestSteps.java]: Done.

File: src/functionaltests/java/com/ericsson/ei/restendpoints/RestEndpointsTestSteps.java (hunk 3 above)
Comment (a-breid-ericsson) [reply src/functionaltests/java/com/ericsson/ei/restendpoints/RestEndpointsTestSteps.java]: Done.

File: src/functionaltests/java/com/ericsson/ei/restendpoints/RestEndpointsTestSteps.java (hunk 4 above)
Comment (a-breid-ericsson) [reply src/functionaltests/java/com/ericsson/ei/restendpoints/RestEndpointsTestSteps.java]: Done.

**Pull request merged**